WooCommerce/Classes/ViewRelated/Themes/ThemesCarouselView.swift

```diff
@@ -0,0 +1,130 @@
+
+    /// The content of the message at the end of the carousel.
+    let lastMessageContent: String
+
+
+    /// Scale of the view based on accessibility changes
+    @ScaledMetric private var scale: CGFloat = 1.0
+
+    var body: some View {
+        ScrollView(.horizontal, showsIndicators: false) {
+            HStack() {
+
+                // Demo for correct state
+                ForEach(themesInfo, id: \.name) { theme in
+                    if let themeUrl = getThemeUrl(themeUrl: theme.url) {
+                        ThemeImage(url: themeUrl)
+                    }
+                }
+
+                // Demo for error state
+                ThemeName(name: themesInfo[0].name)
+
+                // Demo for last message content
+                VStack {
+                    Text(lastMessageHeading)
+                        .bold()
+                        .bodyStyle()
+                        .padding(.horizontal, Layout.contentPadding)
+
+                    Spacer()
+                        .frame(height: Layout.contentPadding)
+
+                    Text(lastMessageContent)
+                        .subheadlineStyle()
+                            .multilineTextAlignment(.center)
+                        .padding(.horizontal, Layout.contentPadding)
+                }.frame(
+                    width: Layout.imageWidth * scale,
+                    height: Layout.imageHeight * scale
+                )
+            }
+        }
+    }
+}
+
+private struct ThemeImage: View {
+    let url: URL
+    @ScaledMetric private var scale: CGFloat = 1.0
+
+    var body: some View {
+        KFImage(url)
+            .resizable()
+            .aspectRatio(contentMode: .fit)
+            .frame(width: Layout.imageWidth * scale, height: Layout.imageHeight * scale)
+            .cornerRadius(Layout.cornerRadius)
+            .shadow(radius: Layout.shadowRadius, x: 0, y: Layout.shadowYOffset)
+            .padding(.init(
+                top: Layout.imageVerticalPadding,
+                leading: Layout.imageHorizontalPadding,
+                bottom: Layout.imageVerticalPadding,
+                trailing: Layout.imageHorizontalPadding)
+            )
+    }
+}
+
+private struct ThemeName: View {
+    let name: String
+
+    var body: some View {
+        VStack {
+            Text(name)
+        }
+        .frame(width: Layout.imageWidth, height: Layout.imageHeight)
+        .background(Color.withColorStudio(
+            name: .wooCommercePurple,
+            shade: .shade0))
+        .cornerRadius(Layout.cornerRadius)
+        .shadow(radius: Layout.shadowRadius, x: 0, y: Layout.shadowYOffset)
+        .padding(.init(
+            top: Layout.imageVerticalPadding,
+            leading: Layout.imageHorizontalPadding,
+            bottom: Layout.imageVerticalPadding,
+            trailing: Layout.imageHorizontalPadding)
+        )
+    }
+}
+
+private extension ThemesCarouselView {
+    private func getThemeUrl(themeUrl: String) -> URL? {
```

Comment:
Nit: How about naming this to `getThemeThumbnailURL` to clarify the intention?

Reply:
I'd suggest adding an extension method for `WordPressTheme` to get this detail as it looks to be fixed:

```Swift
extension WordPressTheme {
    var themeThumbnailURL: URL? {
         // compute the URL and return it
    }
}
```

Reply:
I added the extension method in https://github.com/woocommerce/woocommerce-ios/pull/11336/commits/9a7b191244786cd0a575fa8f8ec51ff2e3d34d42

It's not currently in use, though (since the UI still uses hardcoded value instead of actually using `WordPressTheme`), but I've left a todo comment to remind us to use the real method and delete that `getThemeUrl` function.